src/folding/circuits/hypernova/utils.rs

```diff
@@ -0,0 +1,129 @@
+use crate::folding::circuits::utils::VecFpVar;
+use ark_ff::PrimeField;
+use ark_r1cs_std::{
+    alloc::{AllocVar, AllocationMode},
+    fields::{fp::FpVar, FieldVar},
+    ToBitsGadget,
+};
+use ark_relations::r1cs::{Namespace, SynthesisError};
+use std::{borrow::Borrow, marker::PhantomData};
+
+/// `GammaVar` is a wrapper around an `FpVar` for $\gamma$.
+#[derive(Clone)]
+pub struct GammaVar<F: PrimeField> {
+    pub value: FpVar<F>,
+}
+
+impl<F: PrimeField> AllocVar<F, F> for GammaVar<F> {
+    fn new_variable<T: Borrow<F>>(
+        cs: impl Into<Namespace<F>>,
+        f: impl FnOnce() -> Result<T, SynthesisError>,
+        mode: AllocationMode,
+    ) -> Result<Self, SynthesisError> {
+        FpVar::new_variable(cs, f, mode).map(|value| Self { value })
+    }
+}
+
+impl<F: PrimeField> GammaVar<F> {
+    /// An helper to compute $\gamma^{pow}$
+    pub fn pow(&self, pow: FpVar<F>) -> Result<FpVar<F>, SynthesisError> {
+        let pow = pow.to_bits_le()?;
+        self.value.pow_le(&pow)
+    }
+}
```

Comment:
As mentioned in https://github.com/privacy-scaling-explorations/folding-schemes/pull/53/files#r1446223413 , I'm not sure to understand the reason to add the `GammaVar` type wrapper, the proposal would be to get rid of `GammaVar` and its methods.

Reply:
[Done](https://github.com/privacy-scaling-explorations/folding-schemes/pull/53/commits/e3fcb8790d790c25267fcdfb40e820bf76a6aa3e)!